packages/browser-pyright/src/worker.ts

```diff
@@ -1,11 +1,53 @@
 import { BrowserMessageReader, BrowserMessageWriter, createConnection } from 'vscode-languageserver/browser';
-import { PyrightServer } from './browser-server';
 
-// @ts-ignore
-const ctx: Worker = self as any;
+import { BrowserBackgroundAnalysisRunner, PyrightServer } from './browser-server';
+import { InitializationData } from 'pyright-internal/backgroundThreadBase';
+import { initializeWorkersHost } from 'pyright-internal/common/workersHost';
+import { BrowserWorkersHost } from './browserWorkersHost';
 
-const createLanguageServer = new PyrightServer(
-    createConnection(new BrowserMessageReader(ctx), new BrowserMessageWriter(ctx))
-);
+const ctx: DedicatedWorkerGlobalScope & { app: PyrightServer | BrowserBackgroundAnalysisRunner | undefined } =
+    self as any;
 
-export default createLanguageServer;
+interface BootParams {
+    type: 'browser/boot';
+    mode: 'foreground' | 'background';
+    // Background only.
+    initialData?: InitializationData;
+    // Background only.
+    port?: MessagePort;
+}
+
+// Ideally we'd use a nested worker for the background thread but no Safari support.
+// Instead non-Worker code must facilitate the connection by creating a worker and
+// passing on a port.
+ctx.addEventListener('message', (e: MessageEvent) => {
+    if (e.data.type === 'browser/boot') {
+        const params = e.data as BootParams;
+        const { mode, port, initialData } = params;
+        try {
+            if (mode === 'foreground') {
+                console.log('Booting Pyright foreground worker');
```

Comment:
Probably time to remove these. Booting them already results in messages logged over the JSON-RPC connection.